libraries/core/classes/PipelineManager/index.js

```diff
@@ -83,61 +91,92 @@ class PipelineManager {
    * @param {Function} reject Rejects the promise.
    */
   createRequestCallback(serial, resolve, reject) {
-    const { request } = this.requests.get(serial);
+    const entry = this.requests.get(serial);
+    const { request } = entry;
 
-    request.callback = (error, serialResult, output) => {
-      // Map some values to the request instance to be accessible later.
+    // Add the executor functions to the request object.
+    request.resolve = resolve;
+    request.reject = reject;
+
+    /**
+     * A callback that is invoked when a pipeline response comes in.
+     * @param {Object} error A pipeline error object.
+     * @param {string} serialResult A pipeline serial.
+     * @param {Object} output The pipeline response payload.
+     */
+    const callback = (error, serialResult, output) => {
+      // Add the relevant response properties to the request object.
       request.error = error;
       request.output = output;
-      request.resolve = resolve;
-      request.reject = reject;
+
+      entry.finished = true;
 
       if (request.process === processTypes.PROCESS_SEQUENTIAL) {
         this.handleResultSequence();
       } else {
-        this.handleResult(serial);
+        this.handleResult(serialResult);
       }
     };
+
+    // Take care that the callback always has the correct context.
+    request.callback = callback.bind(this);
+
+    const callbackName = request.getEventCallbackName();
+
+    // Register a response listener for the request.
+    event.addCallback(callbackName, request.callback);
   }
 
   /**
    * Checks wether a pipeline request has running dependencies.
-   * @param {string} serial The pipeline request serial.
+   * @param {string} pipelineName The name of the pipeline.
    * @return {boolean}
    */
-  hasRunningDependencies(serial) {
-    const pipelineName = this.getPipelineNameBySerial(serial);
+  hasRunningDependencies(pipelineName) {
     const dependencies = pipelineDependencies.get(pipelineName);
     let found = 0;
 
     dependencies.forEach((dependency) => {
```

Comment:
`Map` only provides `forEach` as a convenient way to loop through all it's entries. `entries()` in combination with a `for...of` loop might be an alternative, since it's breakable.

Reply:
OK. Since for...of loop is illegal according to our linting rules.